hoptimator-util/src/main/java/com/linkedin/hoptimator/util/planner/EngineRules.java

```diff
@@ -0,0 +1,164 @@
+package com.linkedin.hoptimator.util.planner;
+
+import java.util.Collections;
+
+import org.apache.calcite.adapter.jdbc.JdbcConvention;
+import org.apache.calcite.adapter.jdbc.JdbcImplementor;
+import org.apache.calcite.adapter.jdbc.JdbcRel;
+import org.apache.calcite.jdbc.JavaTypeFactoryImpl;
+import org.apache.calcite.linq4j.tree.Expression;
+import org.apache.calcite.plan.Convention;
+import org.apache.calcite.plan.RelOptCluster;
+import org.apache.calcite.plan.RelOptCost;
+import org.apache.calcite.plan.RelOptPlanner;
+import org.apache.calcite.plan.RelOptRule;
+import org.apache.calcite.plan.RelOptTable;
+import org.apache.calcite.plan.RelTraitSet;
+import org.apache.calcite.rel.InvalidRelException;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.convert.ConverterRule;
+import org.apache.calcite.rel.core.Join;
+import org.apache.calcite.rel.core.JoinRelType;
+import org.apache.calcite.rel.core.TableScan;
+import org.apache.calcite.rel.metadata.RelMetadataQuery;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.rel.type.RelDataTypeSystem;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.rex.RexProgram;
+import org.apache.calcite.schema.Table;
+import org.apache.calcite.schema.SchemaPlus;
+import org.apache.calcite.sql.SqlDialect;
+import org.apache.calcite.sql.dialect.AnsiSqlDialect;
+import org.apache.calcite.sql.dialect.MysqlSqlDialect;
+import org.apache.calcite.sql.type.SqlTypeFactoryImpl;
+
+import com.linkedin.hoptimator.Engine;
+
+
+/** Calling convention using a remote execution engine. */
+public final class EngineRules {
+
+  private final Engine engine;
+
+  public EngineRules(Engine engine) {
+    this.engine = engine;
+  }
+
+  public void register(HoptimatorJdbcConvention inTrait, RelOptPlanner planner) {
+    SqlDialect dialect = dialect(engine);
+    String name = engine.engineName() + "-" + inTrait.database();
+    JdbcConvention outTrait = JdbcConvention.of(dialect, inTrait.expression, name);
+ 
+    System.out.println("Registering rules for " + name + " using dialect " + dialect.toString());
```

Comment:
nit: use logger or remove (applicable to other lines in this file as well)

Reply:
will do